src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/SqlAction.cs

```diff
@@ -611,12 +530,14 @@ LockOption lockOption
     {
         await using NpgsqlCommand command =
             new(
-                $@"SELECT  d.ResourceName FROM dms.Document d INNER JOIN (
-                   SELECT ParentDocumentId, ParentDocumentPartitionKey FROM dms.Reference r
-                   INNER JOIN dms.Alias a ON r.ReferentialId = a.ReferentialId AND r.ReferentialPartitionKey = a.ReferentialPartitionKey
-                   INNER JOIN dms.Document d2 ON d2.Id = a.DocumentId AND d2.DocumentPartitionKey = a.DocumentPartitionKey
-                   WHERE d2.DocumentUuid =$1 AND d2.DocumentPartitionKey = $2) AS re
-                   ON re.ParentDocumentId = d.id AND re.ParentDocumentPartitionKey = d.DocumentPartitionKey
+                $@"SELECT d.ResourceName FROM dms.Document d
+                   INNER JOIN (
+                     SELECT ParentDocumentId, ParentDocumentPartitionKey
+                     FROM dms.Reference r
+                     INNER JOIN dms.Document d2 ON d2.Id = r.ReferencedDocumentId
+                       AND d2.DocumentPartitionKey = r.ReferencedDocumentPartitionKey
+                       WHERE d2.DocumentUuid = $1 AND d2.DocumentPartitionKey = $2) AS re
+                     ON re.ParentDocumentId = d.id AND re.ParentDocumentPartitionKey = d.DocumentPartitionKey
```

Comment:
One less join now to get post-delete failure info. All joins are also now on indexes.